src/main/java/seedu/address/model/loan/Nric.java

```diff
@@ -0,0 +1,82 @@
+package seedu.address.model.loan;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Class that stores the NRIC of a person.
+ */
+public class Nric {
+
+    public static final String MESSAGE_NRIC_CONSTRAINTS =
+            "Nric should be singapore issued. It may be blank. ";
+
+    public static final String NRIC_VALIDATION_REGEX = "^[STGF][0-9]{7}[A-Z]$";
+
+    public final String nric;
+
+    /**
+     * Constructs a {@code Nric}.
+     *
+     * @param ic A valid nric.
+     */
+    public Nric(String ic) {
+        ic = ic.trim().toUpperCase();
+        requireNonNull(ic);
+        checkArgument(isValidNric(ic), MESSAGE_NRIC_CONSTRAINTS);
+        nric = ic;
+    }
+
+    /**
+     * Returns true if a given string is a valid nric.
+     */
+    public static boolean isValidNric(String test) {
+        String ic = test.trim().toUpperCase();
+
+        if (!ic.matches(NRIC_VALIDATION_REGEX)) {
+            return false;
+        }
+
+        String prefix = String.valueOf(ic.charAt(0));
+        String checksum = String.valueOf(ic.charAt(8));
+        String digits = ic.substring(1, 8);
+
+        int[] weights = {2, 7, 6, 5, 4, 3, 2};
+        int sum = 0;
+
+        // Generate checksum
+        for (int i = 0; i < digits.length(); i++) {
+            sum += Integer.parseInt(String.valueOf(digits.charAt(i))) * weights[i];
+        }
+
+        // Add 4 to IC issued in 21th century. This rule was designed by the Singapore government.
+        if ("G".equals(prefix) || "T".equals(prefix)) {
+            sum += 4;
+        }
+
+        if ("S".equals(prefix) || "T".equals(prefix)) {
+            String[] nricCheckDigits = {"J", "Z", "I", "H", "G", "F", "E", "D", "C", "B", "A"};
+            return nricCheckDigits[sum % 11].equals(checksum);
```

Comment:
'11' is a magic number. Can use nricCheckDigits.length.

Reply:
I think this is OK, because this number is a constant used as part of the NRIC verification algorithm in Singapore's system. nricCheckDigits.length is based on said "magic number", not the other way around.

Reply:
True, but it's a consideration anyway to make sure `11` and `nricCheckDigits.length` tally. I would extract both of them out into a `final` int but Java doesn't like enforcing statically-sized arrays. :/

Reply:
The algorithm uses % 11. There is no rule that says that says it is "% (nricCheckDigits)". It could have been % 20, and those whose output is more than 11 is invalid.